core/include/scipp/core/element/rebin.h

```diff
@@ -37,13 +60,19 @@ static constexpr auto rebin = overloaded{
           const auto delta = std::min<double>(xn_high, xo_high) -
                              std::max<double>(xn_low, xo_low);
           const auto owidth = xo_high - xo_low;
-          const auto scale = delta / owidth;
-          if constexpr (is_ValueAndVariance_v<
-                            std::decay_t<decltype(data_old)>>) {
+
+          constexpr bool isValAndVariance =
+              is_ValueAndVariance_v<std::decay_t<decltype(data_old)>>;
+
+          if constexpr (isValAndVariance) {
+            // Avoid implicit promotion and demotion when multiplying by scale
+            const auto scale =
+                std::remove_reference_t<decltype(data_new.value[0])>(delta /
+                                                                     owidth);
```

Comment:
This is a very verbose fix. Can't we just change how `delta` is created, e.g., using `std::min(...)` without explicit template arg, or with the desired size?

Reply:
Creating is with desired size rather than the implicit double would be better, as we then get no mismatched type warnings

However how would you recommend naming that type? I personally don't like the verbosity either

Reply:
Let us discuss this after understanding why you had to add the helper function for the other branch.